sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/aio/_azureappconfigurationproviderasync.py

```diff
@@ -194,9 +206,11 @@ def _buildprovider(
     return provider
 
 
-async def _resolve_keyvault_reference(config, provider: "AzureAppConfigurationProvider") -> str:
+async def _resolve_keyvault_reference(
+    config: "SecretReferenceConfigurationSetting", provider: "AzureAppConfigurationProvider"
+) -> str:
     # pylint:disable=protected-access
-    if provider._key_vault_options is None:
+    if not (provider._key_vault_credentials or provider._key_vault_client_configs or provider._secret_resolver):
         raise ValueError("Key Vault options must be set to resolve Key Vault references.")
```

Comment:
This error message still makes sense in the context of the new parameters - though maybe we could add a bit more detail.

Reply:
@annatisch How about the new version?